satellite/nodestats/endpoint.go

```diff
@@ -45,12 +47,14 @@ func (e *Endpoint) GetStats(ctx context.Context, req *pb.GetStatsRequest) (_ *pb
 
 	peer, err := identity.PeerIdentityFromContext(ctx)
 	if err != nil {
-		return nil, NodeStatsEndpointErr.Wrap(err)
+		return nil, status.Error(codes.Unauthenticated, err.Error())
 	}
-
 	node, err := e.overlay.Get(ctx, peer.ID)
 	if err != nil {
-		return nil, NodeStatsEndpointErr.Wrap(err)
+		if overlay.ErrNodeNotFound.Has(err) {
+			return nil, status.Error(codes.PermissionDenied, err.Error())
+		}
+		return nil, status.Error(codes.Internal, err.Error())
```

Comment:
On Internal errors, we want to log for getting as much information as we can on what was wrong.

Reply:
Good point.